spec/functions/clamp_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+describe 'clamp' do
+  it { is_expected.not_to eq(nil) }
+  it { is_expected.to run.with_params().and_raise_error(Puppet::ParseError) }
+  it { is_expected.to run.with_params("string1").and_raise_error(Puppet::ParseError) }
+  it { is_expected.to run.with_params([]).and_raise_error(Puppet::ParseError) }
+  it { is_expected.to run.with_params([4,3,99]).and_return(4) }
+  it { is_expected.to run.with_params(444,3,99).and_return(99) }
+  it { is_expected.to run.with_params([4,873],73).and_return(73) }
+  it { is_expected.to run.with_params([4],8,99).and_return(8) }
+  it { is_expected.to run.with_params([4],[31], 9911).and_return(31) }
```

Comment:
These need to test all allowed inputs and input combinations. Specifically at least all the tests with (mixed) strings, error handling in the case of non-numeric strings, and error handling with more than three elements.